src/test/java/com/rocketden/main/api/RoomTests.java

```diff
@@ -445,6 +450,164 @@ public void changeRoomHostNewHostNotFound() throws Exception {
         assertEquals(ERROR.getResponse(), actual);
     }
 
+    @Test
+    public void updateRoomSettingsSuccess() throws Exception {
+        UserDto host = new UserDto();
+        host.setNickname("host");
+        UserDto user = new UserDto();
+        user.setNickname("test");
+
+        RoomDto room = setUpRoomWithTwoUsers(host, user);
+
+        UpdateSettingsRequest updateRequest = new UpdateSettingsRequest();
+        updateRequest.setInitiator(host);
+        updateRequest.setDifficulty(Difficulty.EASY);
+
+        MvcResult result = this.mockMvc.perform(put(String.format(PUT_ROOM_SETTINGS, room.getRoomId()))
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(Utility.convertObjectToJsonString(updateRequest)))
+                .andDo(print()).andExpect(status().isOk())
+                .andReturn();
+
+        String jsonResponse = result.getResponse().getContentAsString();
+        room = Utility.toObject(jsonResponse, RoomDto.class);
+
+        assertEquals(updateRequest.getDifficulty(), room.getDifficulty());
+
+        // Confirm with a GET that the room has actually been updated in the database
+        result = this.mockMvc.perform(get(GET_ROOM)
+                .param("roomId", room.getRoomId()))
+                .andDo(print()).andExpect(status().isOk())
+                .andReturn();
+
+        jsonResponse = result.getResponse().getContentAsString();
+        RoomDto actual = Utility.toObject(jsonResponse, RoomDto.class);
+
+        assertEquals(updateRequest.getDifficulty(), actual.getDifficulty());
+    }
+
+    @Test
+    public void updateRoomSettingsNullValue() throws Exception {
+        UserDto host = new UserDto();
+        host.setNickname("host");
+        UserDto user = new UserDto();
+        user.setNickname("test");
+
+        RoomDto room = setUpRoomWithTwoUsers(host, user);
+
+        UpdateSettingsRequest updateRequest = new UpdateSettingsRequest();
+        updateRequest.setInitiator(host);
+
+        MvcResult result = this.mockMvc.perform(put(String.format(PUT_ROOM_SETTINGS, room.getRoomId()))
+                .contentType(MediaType.APPLICATION_JSON_VALUE)
+                .content(Utility.convertObjectToJsonString(updateRequest)))
+                .andDo(print()).andExpect(status().isOk())
+                .andReturn();
+
+        String jsonResponse = result.getResponse().getContentAsString();
+        room = Utility.toObject(jsonResponse, RoomDto.class);
+
+        assertNull(room.getDifficulty());
```

Comment:
Should the `room` be assigned a default difficulty? I'm open to whatever on this -- it might help avoid having a bunch of Null Pointer Exceptions in our code, but null might be useful if we want to show an "undecided" state on the frontend.

Reply:
Yeah I was thinking that a null value represents no difficulty selected, as I think that might be a good default esp. if people don't care as much about advanced settings. Lmk your thoughts on this tho

Reply:
Yeah I actually think that's good, you should keep this 👍